libs/utils/trace_analysis.py

```diff
@@ -921,7 +921,38 @@ def _integrate_square_wave(self, sq_wave):
         else:
             return sum(comp_sig.iloc[1::2].index - comp_sig.iloc[:-1:2].index)
 
-    @memoized
+    def _cropToXTimeRange(self, data):
+        """
+        Crop a dataframe containing FTrace events to the X time range specified
+        by the user through setXTimeRange().
+
+        In case self.x_min and self.x_max are set to values different than 0.0
+        and the trace duration, a first and a last entry are inserted in the
+        cropped series, such that:
+
+            - first element is a copy of the one immediately before time
+              self.x_min
+            - last element is a copy of the one immediately before time
+              self.x_max
+
+        :param data: dataframe to be cropped
+        :type data: :mod:`pandas.Series`
+        """
+        if not isinstance(data, pd.Series):
+            raise ValueError('cropping supported only for pandas.Series'\
+                             'objects!')
+
+        first = None
+        if self.x_min > 0.0:
+            first = pd.Series([data[data.index < self.x_min].iloc[-1]],
+                              index=[self.x_min])
+        last = None
+        if self.x_max < self.trace.time_range:
+            last = pd.Series([data[data.index <= self.x_max].iloc[-1]],
+                             index=[self.x_max])
+        data = data.loc[self.x_min:self.x_max]
+        return pd.concat([first, data, last])
```

Comment:
I tested again the implementation with this simplification and I found an issue.

Basically, if we crop the dataframe this way we might lose lose an event and make wrong computations afterwards. For example, assume we have the following data:

```python
     freq
0.1  600
0.2  1200
0.3  600
0.4  1200
```

If I crop the data in the range `(0.15, 0.35)` I'll get the following:

```python
     freq
0.2  1200
0.3  600
```

Two things happened:

1. I lost the information about the first event and don't know any more that frequency at 0.15 was 1200. As a consequence following computations will be incorrect.

2. I know the there are events after 0.35, but because of what I did I lost the information for the last 0.05 seconds.

Now, what you proposed kind of overcomes this loss of information by adding more information which might be a lot out of the specified range, falling once again into wrong computations later on.

Therefore, we need those tricks that add an entry at the beginning and at the end.